services/compute/virtualmachine/client.go

```diff
@@ -88,6 +88,9 @@ func (c *VirtualMachineClient) Restart(ctx context.Context, group string, name s
 	if err != nil {
 		return
 	}
+	//wait for 5 seconds before starting the VM
```

Comment:
Just thinking out loud, I wonder if it makes sense to have a loop that waits for the powerstate to change to OFF before sending the Start request, instead of a blind sleep.

Reply:
We can create a loop to monitor the power state that also has a timeout.